src/Nethermind/Nethermind.Evm/Tracing/EstimateGasTracer.cs

```diff
@@ -194,10 +197,73 @@ internal long CalculateAdditionalGasRequired(Transaction tx, IReleaseSpec releas
             return _currentGasAndNesting.Peek().AdditionalGasRequired + RefundHelper.CalculateClaimableRefund(intrinsicGas + NonIntrinsicGasSpentBeforeRefund, TotalRefund, releaseSpec);
         }
 
-        public long CalculateEstimate(Transaction tx, IReleaseSpec releaseSpec)
+        private static bool Executable(Transaction transaction, BlockHeader block, GethLikeTxTracer txTracer, UInt256 gas, ITransactionProcessor transactionProcessor)
```

Comment:
All this code shouldn't be in a Tracer, Tracer did its job and this should be somewhere else (single responsibility)

Reply:
Created additional class to handle this.